contrib/tools/workflowcheck/determinism/checker.go

```diff
@@ -381,6 +366,42 @@ func (c *collector) collectFuncInfo(fn *types.Func, decl *ast.FuncDecl) {
 	})
 }
 
+func (c *collector) checkRangeType(rangeType types.Type, n ast.Node, fn *types.Func) Reason {
+Loop:
+	// Unwrap named or generic type
+	for {
+		switch rangeType.(type) {
+		case *types.Named, *types.TypeParam:
+			rangeType = rangeType.Underlying()
+		default:
+			break Loop
+		}
+	}
+	switch t := rangeType.(type) {
```

Comment:
```suggestion
	switch t := rangeType.(type) {
	case *types.Named:
		return c.checkRangeType(rangeType.Underlying(), n, fn)
	case *types.TypeParam:
		return c.checkRangeType(rangeType.Underlying(), n, fn)
```

Recursion reads a bit better to me, but not a big deal